pandas/indexes/numeric.py

```diff
@@ -334,7 +312,7 @@ def equals(self, other):
                 return False
             left, right = self._values, other._values
             return ((left == right) | (self._isnan & other._isnan)).all()
-        except TypeError:
+        except (TypeError, ValueError):
```

Comment:
Yes, this can be simplified a lot. But I need a clarification: currently ``Index.equals(self, other)`` returns ``False`` whenever ``other`` is not an ``Index`` (e.g. an array). Instead ``Int64Index`` and ``Float64Index`` don't - they just run the comparison. The [documentation](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Index.equals.html) seems to suggest that ``other`` _must_ be an index... but then I'd rather raise on a non-``Index`` ``other`` than return ``False``, no?